### PR TITLE
Update TADA_FlagContinuousData

### PR DESCRIPTION
**Pull request by cristinamullin**

### CODE REVIEW

File: R/Utilities.R [hunk 1]
```diff
@@ -134,15 +135,50 @@ TADA_AutoClean <- function(.data) {
...
+  # check to make sure columns do not already exist and capitalize fields with known synonyms that only differ in caps
   print("TADA_Autoclean: creating TADA-specific columns.")
-  .data$TADA.CharacteristicName <- toupper(.data$CharacteristicName)
-  .data$TADA.ResultSampleFractionText <- toupper(.data$ResultSampleFractionText)
-  .data$TADA.MethodSpeciationName <- toupper(.data$MethodSpeciationName)
-  .data$TADA.ResultMeasure.MeasureUnitCode <- toupper(.data$ResultMeasure.MeasureUnitCode)
-  .data$TADA.ActivityMediaName <- toupper(.data$ActivityMediaName)
-  .data$TADA.DetectionQuantitationLimitMeasure.MeasureUnitCode <-
-    toupper(.data$DetectionQuantitationLimitMeasure.MeasureUnitCode)
+
```
Comment (hillarymarler) [inline R/Utilities.R:140]: I like these changes!

File: vignettes/TADAModule1.Rmd [hunk 2]
```diff
@@ -75,11 +75,11 @@ remotes::install_github("USEPA/TADA",
 # options(download.file.method = "wininet")
 ```
 
-```{r for-development-only, echo = F, results = 'hide', message = FALSE, warning = FALSE}
+```{r for-development-only, echo = F, eval = F, results = 'hide', message = FALSE, warning = FALSE}
 # IF YOU ARE NOT DEVELOPING TADA, SKIP THIS CODE CHUNK
 # when developing the package, update this chunk to the current repository branch, so it runs with all of the new features prior to a PR to develop
 remotes::install_github("USEPA/TADA",
-  ref = "updates",
+  ref = "contdataupdates_cm",
```
Comment (hillarymarler) [inline vignettes/TADAModule1.Rmd:82]: Change ref back to develop after updates?

**Review by hillarymarler — APPROVED** [review]

File: R/ResultFlagsIndependent.R [hunk 3]
```diff
@@ -131,138 +131,250 @@ TADA_FlagMethod <- function(.data, clean = TRUE, flaggedonly = FALSE) {
...
+  # everything not YET in cont dataframe
   noncont.data <- subset(.data, !.data$ResultIdentifier %in% cont.data$ResultIdentifier)
 
-  # if there is aggregated continuous data is in the data set
-  if (nrow(cont.data) != 0) {
+  # if time field is not NA, find time difference between results
+  if (length(noncont.data) >= 1) {
+    for (i in 1:nrow(noncont.data)) {
+      if (!is.na(noncont.data$ActivityStartDateTime[i])) {
+        if (noncont.data$TADA.ContinuousData.Flag[i] != "Continuous") {
```
[the remark this answers is not in the document]
Comment (cristinamullin) [reply R/ResultFlagsIndependent.R:265]: good catch! I changed some logic during dev and this is no longer needed

File: R/ResultFlagsIndependent.R [hunk 4]
```diff
@@ -131,138 +131,250 @@ TADA_FlagMethod <- function(.data, clean = TRUE, flaggedonly = FALSE) {
...
+      time.taken <- round(end.time - start.time, 2)
+      print(time.taken)
+
       return(flag.data)
     }
 
     # clean output
     if (clean == TRUE & flaggedonly == FALSE) {
       # filter out invalid characteristic-unit-media combinations
-      clean.data <- dplyr::filter(flag.data, !(TADA.AggregatedContinuousData.Flag %in% "Continuous"))
```
[the remark this answers is not in the document]
Comment (cristinamullin) [reply R/ResultFlagsIndependent.R:237]: Unfortunately there isn't a single field to check. We have a new ActivityTypeCode from 2023 ("Field Msr/Obs-Continuous Time Series") which would be easiest (but it is not going to be populated for older data). After that there are a series of metadata fields to look for together. That's what this whole section is doing. I just added one more - we can look for the string "continuous" in the projectID as well.

**Pull request merged**